script/test/DeployEmailRecoveryModule.t.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "forge-std/Test.sol";
+import "forge-std/console2.sol";
+import { DeployEmailRecoveryModuleScript } from "../DeployEmailRecoveryModule.s.sol";
+import { BaseDeployTest } from "./BaseDeployTest.sol";
+
+/**
+ * @title DeployEmailRecoveryModule_Test
+ * @dev Test contract for deploying the Email Recovery Module
+ */
+contract DeployEmailRecoveryModule_Test is BaseDeployTest {
+    /**
+     * @dev Sets up the test environment.
+     */
+    function setUp() public override {
+        super.setUp();
+    }
+
+    /**
+     * @dev Tests that the standard deployment process executes correctly.
+     */
+    function test_run() public {
+        DeployEmailRecoveryModuleScript target = new DeployEmailRecoveryModuleScript();
+        target.run();
+    }
+
+    /**
+     * @dev Tests the deployment process when the VERIFIER environment variable is not set.
+     */
+    function test_run_no_verifier() public {
+        vm.setEnv("VERIFIER", vm.toString(address(0)));
+        DeployEmailRecoveryModuleScript target = new DeployEmailRecoveryModuleScript();
+        target.run();
+    }
+
+    /**
+     * @dev Tests the deployment process when the DKIM_REGISTRY environment variable is not set.
+     */
+    function test_run_no_dkim_registry() public {
+        vm.setEnv("DKIM_REGISTRY", vm.toString(address(0)));
+        DeployEmailRecoveryModuleScript target = new DeployEmailRecoveryModuleScript();
+        target.run();
+    }
+}
+
+/**
+ * @title DeployEmailRecoveryModule_TestFail
+ * @dev Test contract for failure scenarios when deploying the Email Recovery Module
+ */
+contract DeployEmailRecoveryModule_TestFail is BaseDeployTest {
+    /**
+     * @dev Sets up the test environment.
+     */
+    function setUp() public override {
+        super.setUp();
+    }
+
+    /**
+     * @dev Tests that deployment fails when both DKIM_REGISTRY and SIGNER environment variables are
+     * not set.
+     */
+    function testFail_run_no_dkim_registry_no_signer() public {
```

Comment:
`testFail...` may be deprecated in the future https://github.com/foundry-rs/foundry/issues/4437. Don't think it is worth changing here now but FYI